### ISSUE TITLE
Error when using IPAdapter or IPAdapter Embeds post update

### ISSUE THREAD

**Issue by CPapadim** [cubiq/ComfyUI_IPAdapter_plus#370]
I updated IPAdapter (and comfyUI) and rebuilt workflows that were working before using the new nodes.  The workflows are still generally working, but about half the time the workflows give this error:

```RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument mat1 in method wrapper_CUDA_addmm)```

This happens during the `Request to load ControlNet` step of the KSampler.  The workflow doesn't use any special nodes aside from IPAdapter, KSampler(Advanced), and rgthree nodes.  

Things I tried are:
* Bypassing IPAdapter (error goes away completely)
* Fresh install of comfy + ipadapter (error remains)
* Both Embeds and regular image IPadapter (both cause error)
* Both unified loader and manual loading of related models (both cause error)
* I've read that some comfy custom nodes in the past caused this error (e.g. FizzNodes), but I don't have those nodes either in the workflow or installed at all, and those maintainers seem to have fixed their nodes anyways.

Running on 4070 with 8GB VRam, 16GB Ram in case it's due to requirements of new IPAdapter being beefier, but the old version was running fine on this system.   The fact that it works some of the time leads me to believe it could be related to specs, though I don't see any 'out of memory' related errors in the Traceback

The workflows are using IPAdapter Plus (no face stuff etc).

**Comment by CPapadim** [comment]
I ran a few more tests and it seems like the first run after modifying controlnet properties will always work, and subsequent runs will fail.  E.g. if I update the ControlNet strength slightly, or the controlnet input image, it will work for that run.   I wonder if there's some sort of stale data retained between runs.

**Comment by cubiq** [comment]
I know what is happening but I would need the full back trace of the error.

**Comment by CPapadim** [comment]
Here it is.  Impact-Pack shows up here but I tried removing it and the error still happens with a slightly different traceback.  IPAdapter is not in there, so I'm not sure exactly how it's impacting things.

```[rgthree] Using rgthree's optimized recursive execution.
Requested to load ControlNet
Loading 1 new model
  0%|                                                                                                                       | 0/12 [00:00<?, ?it/s]
!!! Exception during processing !!!
Traceback (most recent call last):
  File "/ComfyUI/execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "/ComfyUI/execution.py", line 81, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "/ComfyUI/execution.py", line 74, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "/ComfyUI/nodes.py", line 1403, in sample
    return common_ksampler(model, noise_seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise, disable_noise=disable_noise, start_step=start_at_step, last_step=end_at_step, force_full_denoise=force_full_denoise)
  File "/ComfyUI/nodes.py", line 1339, in common_ksampler
    samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
  File "/ComfyUI/custom_nodes/ComfyUI-Impact-Pack/modules/impact/sample_error_enhancer.py", line 22, in informative_sample
    raise e
  File "/ComfyUI/custom_nodes/ComfyUI-Impact-Pack/modules/impact/sample_error_enhancer.py", line 9, in informative_sample
    return original_sample(*args, **kwargs)  # This code helps interpret error messages that occur within exceptions but does not have any impact on other operations.
  File "/ComfyUI/comfy/sample.py", line 100, in sample
    samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "/ComfyUI/comfy/samplers.py", line 705, in sample
    return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "/ComfyUI/comfy/samplers.py", line 610, in sample
    samples = sampler.sample(model_wrap, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "/ComfyUI/comfy/samplers.py", line 548, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "/BaseVENV/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/ComfyUI/comfy/k_diffusion/sampling.py", line 137, in sample_euler
    denoised = model(x, sigma_hat * s_in, **extra_args)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/samplers.py", line 286, in forward
    out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, model_options=model_options, seed=seed)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/samplers.py", line 273, in forward
    return self.apply_model(*args, **kwargs)
  File "/ComfyUI/comfy/samplers.py", line 270, in apply_model
    out = sampling_function(self.inner_model, x, timestep, uncond, cond, cond_scale, model_options=model_options, seed=seed)
  File "/ComfyUI/comfy/samplers.py", line 250, in sampling_function
    cond_pred, uncond_pred = calc_cond_uncond_batch(model, cond, uncond_, x, timestep, model_options)
  File "/ComfyUI/comfy/samplers.py", line 198, in calc_cond_uncond_batch
    c['control'] = control.get_control(input_x, timestep_, c, len(cond_or_uncond))
  File "/ComfyUI/comfy/controlnet.py", line 182, in get_control
    control = self.control_model(x=x_noisy.to(dtype), hint=self.cond_hint, timesteps=timestep.float(), context=context.to(dtype), y=y)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/cldm/cldm.py", line 287, in forward
    emb = self.time_embed(t_emb)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/container.py", line 215, in forward
    input = module(input)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/ops.py", line 52, in forward
    return super().forward(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument mat1 in method wrapper_CUDA_addmm)```

**Comment by cubiq** [comment]
do you mind sending also the back trace without impact?

**Comment by CPapadim** [comment]
Here it is.  A note that may be a red herring, it's harder to reproduce without impact pack, I had to stack two controlnets to do it.  I haven't fully tested this setup so I'm not sure if it has the same behavior as the setup with Impact Pack that I described.  I'll test it and see.

```Requested to load ControlNet
Loading 1 new model
  0%|                                                                                                                                                       | 0/12 [00:00<?, ?it/s]
!!! Exception during processing !!!
Traceback (most recent call last):
  File "/ComfyUI/execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "/ComfyUI/execution.py", line 81, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "/ComfyUI/execution.py", line 74, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "/ComfyUI/nodes.py", line 1403, in sample
    return common_ksampler(model, noise_seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise, disable_noise=disable_noise, start_step=start_at_step, last_step=end_at_step, force_full_denoise=force_full_denoise)
  File "/ComfyUI/nodes.py", line 1339, in common_ksampler
    samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
  File "/ComfyUI/comfy/sample.py", line 100, in sample
    samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "/ComfyUI/comfy/samplers.py", line 705, in sample
    return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "/ComfyUI/comfy/samplers.py", line 610, in sample
    samples = sampler.sample(model_wrap, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "/ComfyUI/comfy/samplers.py", line 548, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "/BaseVENV/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/ComfyUI/comfy/k_diffusion/sampling.py", line 137, in sample_euler
    denoised = model(x, sigma_hat * s_in, **extra_args)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/samplers.py", line 286, in forward
    out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, model_options=model_options, seed=seed)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/samplers.py", line 273, in forward
    return self.apply_model(*args, **kwargs)
  File "/ComfyUI/comfy/samplers.py", line 270, in apply_model
    out = sampling_function(self.inner_model, x, timestep, uncond, cond, cond_scale, model_options=model_options, seed=seed)
  File "/ComfyUI/comfy/samplers.py", line 250, in sampling_function
    cond_pred, uncond_pred = calc_cond_uncond_batch(model, cond, uncond_, x, timestep, model_options)
  File "/ComfyUI/comfy/samplers.py", line 198, in calc_cond_uncond_batch
    c['control'] = control.get_control(input_x, timestep_, c, len(cond_or_uncond))
  File "/ComfyUI/comfy/controlnet.py", line 182, in get_control
    control = self.control_model(x=x_noisy.to(dtype), hint=self.cond_hint, timesteps=timestep.float(), context=context.to(dtype), y=y)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/cldm/cldm.py", line 287, in forward
    emb = self.time_embed(t_emb)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/container.py", line 215, in forward
    input = module(input)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/ComfyUI/comfy/ops.py", line 52, in forward
    return super().forward(*args, **kwargs)
  File "/BaseVENV/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument mat1 in method wrapper_CUDA_addmm)```

**Comment by cubiq** [comment]
mh this is weird, can you also provide the workflow?

**Comment by CPapadim** [comment]
I'll try to create a minimal workflow that reproduces it instead of the beast I have and post it here in a bit

**Comment by CPapadim** [comment]
I'm going to go ahead and close this issue.  In creating a debug version of the workflow I was able to reproduce this issue without IPAdapter.  It has something to do with anything that loads multiple models - with the more models being loaded the more chance for the issue to show up.  The first run always works, but subsequent runs cause the issue if parameters aren't changed.   IPAdapter seems to have just been particularly good at triggering this probably because it loads a bunch of models, but I was able to get this to happen without IPAdapter.

For example, if I use a combine conditioning and feed that into my controlnet chain so that they are now requested twice, this will also cause the problem (4x controlnet requests in that case).  I still don't understand what's going on, why the first run works but subsequent runs don't, but it seems like it's something to do with doing a run that uses a lot of models, and then doesn't update parameters so that there's some caching going on.  I'll try to isolate further, see if i can get it to happen with just raw ComfyUI and take it from there.  Thanks!

**Comment by cubiq** [comment]
the first time all models and embeds are in CPU, during the generation they are moved to the GPU. Some node doesn't bring them back to the CPU before processing

**Comment by nathanshipley** [comment]
In case it helps someone, I got a similar (but not exactly the same) error when I had forgotten to hook up the model from the Unified Loader to the IPAdapter Advanced node as shown below.

Error:
`Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument weight in method wrapper_CUDA__native_group_norm)`

Causes error:
![image](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/14170135/654d19f1-bcc3-4779-a4ab-d808555fb740)

Then when I hook it up like this, it works:
![image](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/14170135/e33e0c20-1b51-4e57-ae49-d1cd9905799b)

Then, strangely, after queuing the workflow again with it hooked up **incorrectly**, it then executes without an error.  ¯\_(ツ)_/¯

Full trace:
```
Error occurred when executing KSampler:

Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument weight in method wrapper_CUDA__native_group_norm)

File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/nodes.py", line 1369, in sample
return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/nodes.py", line 1339, in common_ksampler
samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-Impact-Pack/modules/impact/sample_error_enhancer.py", line 22, in informative_sample
raise e
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-Impact-Pack/modules/impact/sample_error_enhancer.py", line 9, in informative_sample
return original_sample(*args, **kwargs) # This code helps interpret error messages that occur within exceptions but does not have any impact on other operations.
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/sampling.py", line 372, in motion_sample
latents = wrap_function_to_inject_xformers_bug_info(orig_comfy_sample)(model, noise, *args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/utils_model.py", line 377, in wrapped_function
return function_to_wrap(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-Advanced-ControlNet/adv_control/control_reference.py", line 46, in refcn_sample
return orig_comfy_sample(model, *args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/sample.py", line 100, in sample
samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 705, in sample
return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 610, in sample
samples = sampler.sample(model_wrap, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 548, in sample
samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/k_diffusion/sampling.py", line 580, in sample_dpmpp_2m
denoised = model(x, sigmas[i] * s_in, **extra_args)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 286, in forward
out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, model_options=model_options, seed=seed)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 273, in forward
return self.apply_model(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 270, in apply_model
out = sampling_function(self.inner_model, x, timestep, uncond, cond, cond_scale, model_options=model_options, seed=seed)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/sampling.py", line 412, in evolved_sampling_function
cond_pred, uncond_pred = calc_cond_uncond_batch_wrapper(model, cond, uncond_, x, timestep, model_options)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/sampling.py", line 577, in calc_cond_uncond_batch_wrapper
return comfy.samplers.calc_cond_uncond_batch(model, cond, uncond, x_in, timestep, model_options)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-TiledDiffusion/.patches.py", line 4, in calc_cond_uncond_batch
return calc_cond_uncond_batch_original_tiled_diffusion_413c446b(model, cond, uncond, x_in, timestep, model_options)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/samplers.py", line 224, in calc_cond_uncond_batch
output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/model_base.py", line 97, in apply_model
model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/ldm/modules/diffusionmodules/openaimodel.py", line 850, in forward
h = forward_timestep_embed(module, h, emb, context, transformer_options, time_context=time_context, num_video_frames=num_video_frames, image_only_indicator=image_only_indicator)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/sampling.py", line 102, in forward_timestep_embed
x = layer(x, context)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/motion_module_ad.py", line 616, in forward
return self.temporal_transformer(input_tensor, encoder_hidden_states, attention_mask, self.view_options)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/motion_module_ad.py", line 765, in forward
hidden_states = self.norm(hidden_states).to(hidden_states.dtype)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/comfy/ops.py", line 94, in forward
return super().forward(*args, **kwargs)
File "/home/nathan/AI/code/12_ComfyUI/ComfyUI_09_2024-04-01-IPAv2/custom_nodes/ComfyUI-AnimateDiff-Evolved/animatediff/sampling.py", line 141, in groupnorm_mm_forward
input = group_norm(input, self.num_groups, weight, bias, self.eps)
File "/home/nathan/anaconda3/envs/comfyui/lib/python3.10/site-packages/torch/nn/functional.py", line 2530, in group_norm
return torch.group_norm(input, num_groups, weight, bias, eps, torch.backends.cudnn.enabled)
```

**Comment by CPapadim** [comment]
I was able to verify that my issue was not related to this or IPAdapter - was able to reproduce it with just ComfyUI and no nodes installed . I was further able to get it to go away by reverting to a previous ComfyUI commit before the new model loading and unloading optimizations.  So at least my issue is unrelated to IPAdapter and definitely related to some of the commits made over the last month to ComfyUI to improve model loading.  I opened an [issue there](https://github.com/comfyanonymous/ComfyUI/issues/3179) so hopefully it will be tackled eventually.
